src/git/GitAlert.cpp

```diff
@@ -148,8 +148,6 @@ GitAlert::MessageReceived(BMessage* message)
 			BWindow::MessageReceived(message);
 		} return;
 	}
-	
-	// TODO: Dead code
 	delete_sem(fAlertSem);
 	fAlertSem = -1;
```

Comment:
yes, correct.
The reason is that it was borrowed from QuitAlert where the cancel button directly sends a B_QUIT_REQUESTED message instead of going through the class' MessageReceived.
It was created long before GAlert, it is a standalone class that has only one button so it quits the window when Ok button is pressed. No other buttons are involved.
Long story short, it must be redesigned to inherit from GAlert that is instead designed correctly.